### PR TITLE
Chrome: Clear Post edits on save success

### PR DESCRIPTION
**Pull request by youknowriad**
Currently, we're clearing post edits when we trigger the post save request, this has the drawback of reverting the post state for a small period of time (try switching to private, you'll notice a flickering in the visibility while saving).

This PR moves clearing post edits to the save success handler, but at the same time, if we made other edits while saving, these new edits are kept in the store.

Related https://github.com/WordPress/gutenberg/pull/945#discussion_r120607125

### CODE REVIEW

**Comment by aduth** [comment]
Curious if another option might work better: Rather than waiting 'til after the save and deciding then which keys should be omitted from `edits`, should we optimistically assume that the current saved value of the post should inherit those edited values as the "new truth", and continue as we were to clear edits immediately at the time of save? Becomes a bit tricky to manage if the save attempt fails, but we might be able to store a reference to the original post in the effect handler and revert back to it in such a case.

In any case, if we keep as is, can we add an inline comment reflecting this intent:

>if we made other edits while saving, these new edits are kept in the store.

**Comment by youknowriad** [comment]
I think the solution here is simpler and works just as good. I've added the comment. Merging

**Comment by youknowriad** [comment]
Oh! Sorry for the confusion, I probably dismissed this feedback quickly. let me rephrase this. What benefits do you see in optimistically update the currentPost?

My thinking here is:

 - We'll need another reducer to store the previous saved post to make sure we could restore it if it fails.
 - Minor: What if another save action is triggered, should we store multiple previous posts? I know the UI forbids this for now.
 - I'm finding the current solution really simple and works great. We may be concerned about performance but I don't think we should because this is executed after an async request which in all cases would take way more time and this time is negligible.

**Comment by aduth** [comment]
Thanks for elaborating. My thinking was primarily two-fold:

- Whether we generally treat the flow of data as optimistic or pessimistic. I'd argue the approach here is pessimistic: we wait until we hear back from the server before updating our representation of the post.
- It's no longer simple when our action was previously "clear edits", and is now "clear edits, but only of these key value pairs where the value matching what we'd known to be the edit is _still_ the edited value, otherwise keep the rest"

Not that optimistic updates aren't challenging. There are options though. [`redux-optimist`](https://github.com/ForbesLindesay/redux-optimist) looks interesting for this use case.

**Comment by youknowriad** [comment]
This library looks awesome, I'll give it a try 👍

**Comment by youknowriad** [comment]
replaced by #1093
